### DISCUSSION TITLE
[FR] Show entity picture on sub-buttons

### DISCUSSION THREAD

**Discussion by oerix** [Clooos/Bubble-Card#1007] [Feature requests]
**Describe the bug**  
When using a `person` entity for sub-buttons the user icon is shown, and not the entity picture. It works fine for the main popup header, just not for sub-buttons. I'm not sure if this is a bug or just not supported.

**To Reproduce**  
1. Create Popup Bubble card
2. Select any Button type
3. Add `Person` entity to sub-button field 
4. Toggle `show icon`

**Expected behavior**  
Show person entity picture instead of user icon. Maybe add `show_entity_picture: true` option?

**Screenshots**  
<img width="497" alt="Screenshot 2024-07-11 at 20 44 35" src="https://github.com/Clooos/Bubble-Card/assets/161945818/0ddbdf1c-1632-4561-8bdf-1bf982d0e5f5">

**Informations (please complete the following information):**  
 - OS: macOS
 - Browser/App: Safari, Chrome
 - Bubble Card version: 2.1.0beta-3
 - Home Assistant version: 2024.7.2

Thank you! 🍻

**Comment by oerix** [comment]
Any reason why this has been converted to a discussion? @MrBearPresident

**Comment by MrBearPresident** [comment]
Because this is something that is not supported yet. And so it is an Feature Request.

**Comment by Menelao147** [comment]
You can achieve your goal with a bit of custom styles.

![image](https://github.com/user-attachments/assets/f05602df-10c5-4112-8cff-325c79414997)


This will remove the icon from your SubButton and replace it with the entity picture

```
${subButtonIcon[0].setAttribute("icon", '')}
.bubble-sub-button-1 {
  background-image: url(${hass.states['person.roberto'].attributes.entity_picture});
  background-size: cover;
}
```

**Reply by oerix** [reply]
Thanks for sharing the workaround, I tried your code but it shows blank even after changing the `person.name`?

It would be nice if this was supported natively like HA does.

**Reply by Menelao147** [reply]
> Thanks for sharing the workaround, I tried your code but it shows blank even after changing the `person.name`?
> 
> It would be nice if this was supported natively like HA does.

Could you share your code, so I can test your configuration?

**Reply by oerix** [reply]
> > Thanks for sharing the workaround, I tried your code but it shows blank even after changing the `person.name`?
> > It would be nice if this was supported natively like HA does.
> 
> Could you share your code, so I can test your configuration?

https://dpaste.org/jwwCE

**Reply by Menelao147** [reply]
Okay, it looks like I missed the card container background.
With this approach, it should work:

```
${subButtonIcon[0].setAttribute("icon", '')}
.bubble-button-card-container { background-color: rgba(0,0,0,0) !important; }
.bubble-sub-button-1 {
   background-image: url(${hass.states['person.roberto'].attributes.entity_picture});
   background-size: cover;
}
```

___
<sub>If this solves your problem, mark this as solved and close the discussion.</sub>

**Comment by aferrato** [comment]
It would be great to have native support for displaying the entity picture in sub-buttons. 
This would also be very useful for media_player entities.

thanks.
